projects/igniteui-angular/src/lib/test-utils/grid-base-components.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { ChangeDetectorRef, Component, Input, OnInit, ViewChild } from '@angular/core';
+import { IgxColumnComponent } from '../grid/column.component';
+import { IgxGridComponent } from '../grid/grid.component';
+import { SampleTestData } from './sample-test-data.spec';
+import { ColumnDefinitions, GridTemplateStrings } from './template-strings.spec';
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data">
+        </igx-grid>
+    `
+})
+export class BasicGridComponent {
+    @Input()
+    public data = [];
+
+    @ViewChild(IgxGridComponent)
+    public grid: IgxGridComponent;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate">
+        </igx-grid>
+    `
+})
+export class GridAutoGenerateComponent extends BasicGridComponent {
+    @Input()
+    public autoGenerate = true;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate"
+            [height]="height" [width]="width">
+        </igx-grid>
+    `
+})
+export class GridWithSizeComponent extends GridAutoGenerateComponent {
+    @Input()
+    public width: string;
+
+    @Input()
+    public height: string;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate"
+            [height]="height" [width]="width">
+        </igx-grid>
+    `
+})
+export class GridNxMComponent extends GridWithSizeComponent implements OnInit {
+    public colsCount: number;
+    public rowsCount: number;
+    public columnsType = 'string';
+    public hasEditableColumns = false;
+    public startFromOne = false;
+    public columnNamePrefix = 'col';
+    public columns = [];
+
+    ngOnInit() {
+        this.columns = (this.hasEditableColumns) ? SampleTestData.generateEditableColumns(this.colsCount, this.columnNamePrefix)
+                                : SampleTestData.generateColumnsByType(this.colsCount, this.columnsType, this.columnNamePrefix);
```

Comment:
The hasEditableColumns property has not reflect on the generated columns, because the **editable** property which is part of the generated column object never reaches the defined grid template.